src/AutoMapper/ResolutionContext.cs

```diff
@@ -1,21 +1,18 @@
 using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Linq.Expressions;
-using AutoMapper.Execution;
 
 namespace AutoMapper
 {
     /// <summary>
     /// Context information regarding resolution of a destination value
     /// </summary>
-    public class ResolutionContext : IRuntimeMapper
+    public class ResolutionContext : IInternalRuntimeMapper
     {
         private Dictionary<ContextCacheKey, object> _instanceCache;
         private Dictionary<TypePair, int> _typeDepth;
-        private readonly IRuntimeMapper _inner;
+        private readonly IInternalRuntimeMapper _inner;
 
-        public ResolutionContext(IMappingOperationOptions options, IRuntimeMapper mapper)
+        internal ResolutionContext(IMappingOperationOptions options, IInternalRuntimeMapper mapper)
```

Comment:
If we must, we can keep this public and make `IInternalRuntimeMapper` public.

Reply:
Meh.